rclpy/test/test_node.py

```diff
@@ -838,6 +838,119 @@ def test_node_set_parameters_rejection(self):
         self.assertIsInstance(result[0], SetParametersResult)
         self.assertFalse(result[0].successful)
 
+    def reject_parameter_callback_1(self, parameter_list):
+        rejected_parameters = (param for param in parameter_list if 'reject1' in param.name)
+        return SetParametersResult(successful=(not any(rejected_parameters)))
+
+    def reject_parameter_callback_2(self, parameter_list):
+        rejected_parameters = (param for param in parameter_list if 'reject2' in param.name)
+        return SetParametersResult(successful=(not any(rejected_parameters)))
+
+    def reject_parameter_callback_3(self, parameter_list):
+        rejected_parameters = (param for param in parameter_list if 'reject3' in param.name)
+        return SetParametersResult(successful=(not any(rejected_parameters)))
+
+    def test_node_set_paramters_rejection_list(self):
+        # Declare a new paramters and set a list of callbacks so that it's rejected when set.
+        reject_list_parameter_tuple1 = [
+            (
+                'reject1',
+                True,
+                ParameterDescriptor()
+            ),
+            (
+                'non_reject',
+                True,
+                ParameterDescriptor()
+            ),
+            (
+                'non_reject',
+                True,
+                ParameterDescriptor()
+            )
+        ]
+
+        self.node.declare_parameters('', reject_list_parameter_tuple1)
+        self.node.add_on_set_parameters_callback(self.reject_parameter_callback_1)
+        self.node.add_on_set_parameters_callback(self.reject_parameter_callback_2)
+        self.node.add_on_set_parameters_callback(self.reject_parameter_callback_3)
+
+        result = self.node.set_parameters(
+            [
+                Parameter(
+                    name=reject_list_parameter_tuple1[0][0],
+                    value=reject_list_parameter_tuple1[0][1]
+                ),
+                Parameter(
+                    name=reject_list_parameter_tuple1[1][0],
+                    value=reject_list_parameter_tuple1[1][1]
+                ),
+                Parameter(
+                    name=reject_list_parameter_tuple1[2][0],
+                    value=reject_list_parameter_tuple1[2][1]
+                )
+            ]
+        )
+        self.assertIsInstance(result, list)
+        self.assertIsInstance(result[0], SetParametersResult)
+        self.assertFalse(result[0].successful)
```

Comment:
Would be nice to also test the "happy" path (in other words, verify that you can insert parameters).

Reply:
@allenh1 sir you mean to say if I can declare new parameters and update its values?

Reply:
> if I can declare new parameters and update its values?

Yes

Reply:
@allenh1 is it okay sir?

Reply:
@allenh1 @dirk-thomas sir is it okay now?

Reply:
I don't think this been resolved. Please add another test where we add multiple callbacks, but no parameters are rejected.